lib/ts/recipe/oauth2/types.ts

```diff
@@ -122,3 +151,9 @@ export type OAuth2ClientOptions = {
 };
 
 export type CreateOAuth2ClientInput = Partial<Omit<OAuth2ClientOptions, "createdAt" | "updatedAt">>;
+
+export type UpdateOAuth2ClientInput = Nullable<CreateOAuth2ClientInput>;
```

Comment:
Why do we need to make this `Nullable`? I thought it is already nullable wherever it is a valid value.

Reply:
I have updated the types as the per the Hydra API.

For `OAuth2ClientOptions`, there are 3 categories of properties -

1. Always has a non-nullable value.
2. Always exists (either valid/null value)
3. Only exists if a valid value is given


`CreateOAuth2ClientInput` type is basically `OAuth2ClientOptions` with every field being marked as optional.


For `UpdateOAuth2ClientInput`, we wanted to allowed `null` only for the properties that can be set to `null` via a PATCH request. There were only 9 such properties.